design/baremetal-operator/bmh_non-bootable_iso.md

```diff
@@ -0,0 +1,113 @@
+# Add non-bootable iso attach API to BareMetalHost
+
+## Status
+
+Implementable
+
+## Summary
+
+Add a new field in BareMetalHost so that it is possible to attach a generic
+non-bootable iso image via Ironic, after the host has been provisioned.
+
+## Motivation
+
+In certain scenarios it is desirable to attach a non-bootable iso image
+to a system after provisioning has finished, for example :
+
+A configuration ISO can be used to provide first-boot configuration to a host
+deployed from a standard image in advance. An operator may have a pool of
+powered off hosts with operating system already provisioned. Then, adding
+a host to the cluster boils down to powering it on with a configuration ISO provided.
+
+### Goals
+
+Expose the Ironic API to attach non-bootable iso images via Metal3.
+
+### Non-Goals
+
+Supporting non-ISO images such as a USB stick.
+
+## Proposal
+
+### BMO Proposal
+
+Add a new optional section in the BareMetalHost spec containing details of the
+non-bootable iso image, example :
+
+```yaml
+  spec:
+    dataImage:
+      url: http://1.2.3.4/image.iso
+      credentialsName: dataimage-secret
```

Comment:
The current Ironic patch dropped credentials and disableCertificateVerification because of complexity with their implementation. I suggest we mark this example as the desired future state, but for now only propose the "url" field.

Reply:
Added current design section, and desired future design.